### ISSUE TITLE
sls plugin install fails for unrecognized configuration variable sources

### ISSUE THREAD

**Issue by kyontan** [serverless/serverless#9578]
When I tried to install plugin by `sls plugin install -n serverless-python-requirements`, it fails for following error message:

```
ServerlessError: Approached unrecognized configuration variable sources: "self"
```

I expect that plugin installation process does not need to expand variable sources like `self` or `file`.

(But I understand parsing `serverless.yml` is essential process because `sls install` appends `plugins:` section to the file)

As a workaround, I commented out all usage of variable sources like `${self:provider.stage}`, and it got works.

<details>
<summary><code>serverless.yml</code></summary>

```yaml
service: example-function
frameworkVersion: '2'
variablesResolutionMode: 20210326
configValidationMode: error

package:
  patterns:
    - '!./**'
    - ./*.py
    - ./requirements.txt

provider:
  name: aws
  runtime: python3.8
  lambdaHashingVersion: 20201221

  iam:
    role:
      statements:
        - Effect: "Allow"
          Action:
            - "logs:CreateLogGroup"
            - "logs:CreateLogStream"
            - "logs:PutLogEvents"
          Resource:
            - "arn:aws:logs:*:*:*"
        - Effect: "Allow"
          Action:
            - "ec2:CreateNetworkInterface"
            - "ec2:DescribeNetworkInterfaces"
            - "ec2:DeleteNetworkInterface"
          Resource:
            - "*"

functions:
  example-function:
    handler: handler.lambda_handler
    name: sls-${self:provider.stage}-example-function
```

</details>

<details>
<summary><b><code>sls plugin install -n serverless-python-requirements</code> output</b></summary>

```
# SLS_DEBUG="*" sls plugin install -n serverless-python-requirements
Serverless: Load command interactiveCli
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command config:tabcompletion
Serverless: Load command config:tabcompletion:install
Serverless: Load command config:tabcompletion:uninstall
Serverless: Load command create
Serverless: Load command install
Serverless: Load command package
Serverless: Load command deploy
Serverless: Load command deploy:function
Serverless: Load command deploy:list
Serverless: Load command deploy:list:functions
Serverless: Load command invoke
Serverless: Load command invoke:local
Serverless: Load command info
Serverless: Load command logs
Serverless: Load command metrics
Serverless: Load command print
Serverless: Load command remove
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Load command slstats
Serverless: Load command plugin
Serverless: Load command plugin
Serverless: Load command plugin:install
Serverless: Load command plugin
Serverless: Load command plugin:uninstall
Serverless: Load command plugin
Serverless: Load command plugin:list
Serverless: Load command plugin
Serverless: Load command plugin:search
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command upgrade
Serverless: Load command uninstall
Serverless: Load command requirements
Serverless: Load command requirements:clean
Serverless: Load command requirements:install
Serverless: Load command requirements:cleanCache
Serverless: Load command login
Serverless: Load command logout
Serverless: Load command generate-event
Serverless: Load command test
Serverless: Load command dashboard
Serverless: Load command output
Serverless: Load command output:get
Serverless: Load command output:list
Serverless: Load command param
Serverless: Load command param:get
Serverless: Load command param:list
Serverless: Load command studio

 Serverless Error ----------------------------------------

  ServerlessError: Approached unrecognized configuration variable sources: "self"
      at /Users/me/.nodebrew/node/v15.13.0/lib/node_modules/serverless/scripts/serverless.js:636:17
      at async /Users/me/.nodebrew/node/v15.13.0/lib/node_modules/serverless/scripts/serverless.js:450:7

  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com

  Your Environment Information ---------------------------
     Operating System:          darwin
     Node Version:              15.13.0
     Framework Version:         2.45.1
     Plugin Version:            5.3.0
     SDK Version:               4.2.3
     Components Version:        3.12.0
```

</details>

<b>Installed version</b>

```
# serverless --version
Framework Core: 2.45.1
Plugin: 5.3.0
SDK: 4.2.3
Components: 3.12.0
```

**Comment by medikoo** [comment]
@kyontan great thanks for reporting that! I've reproduced it locally, fix will be introduced shortly

**Comment by kyontan** [comment]
@medikoo Thanks for your blazing fast ⚡ fix & release. 

I confirmed it was fixed in 2.45.2. 

Thanks!
